### PR TITLE
Add api doc for programmatic animation classes

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by jbogarthyde** [comment]
@matsko Please review content, and help me fix the golden file so it will pass the circleci:test.

File: packages/animations/src/animation_builder.ts [hunk 1]
```diff
@@ -9,54 +9,68 @@ import {AnimationMetadata, AnimationOptions} from './animation_metadata';
...
- * Programmatic animations are first built and then a player is created when the build animation is
- * attached to an element.
+ * To use this service, add it to your component or directive as a dependency.
+ * The service is instantiated along with your component. Access the instance
+ * through the `_builder` property.
+ *
+ * * Use the `build()` method to create a programmatic animation using the
+ * `animate()` function. The method returns an `AnimationFactory` instance.
+ * * Use the factory object to create an `AnimationPlayer` and attach it to a DOM element.
+ * * Use the player object to control the animation programmatically.
```
Comment (jenniferfell) [inline packages/animations/src/animation_builder.ts]: I'm unclear about the relationship among bullet items. Add intro sentence. If these are a sequence, use numbers. 
I wasn't clear if they were all the steps or options until I looked at the code. (I really like the synergy between these and the code comments.)

File: packages/animations/src/animation_builder.ts [hunk 2]
```diff
@@ -9,54 +9,68 @@ import {AnimationMetadata, AnimationOptions} from './animation_metadata';
...
- * An instance of `AnimationFactory` is returned from {@link AnimationBuilder#build
- * AnimationBuilder.build}.
+ * A factory object returned from the `AnimationBuilder`.`build()` method.
  *
- * @experimental Animation support is experimental.
  */
 export abstract class AnimationFactory {
+  /**
+   * Creates an `AnimationPlayer` instance for the reusable animation defined by
+   * the `AnimationBuilder`.`build()` method that returned this instance,
```
Comment (jenniferfell) [inline packages/animations/src/animation_builder.ts]: Unclear parsing "that returned this instance"? Creates an instance for the thing defined by a method that returned this same instance?

File: packages/animations/src/animation_builder.ts [hunk 3]
```diff
@@ -9,54 +9,68 @@ import {AnimationMetadata, AnimationOptions} from './animation_metadata';
...
+   * and attaches to a DOM element.
+   * @param element The DOM element to which to attach the animation.
+   * @param options A set of options that can include a time delay, and
```
Comment (jenniferfell) [inline packages/animations/src/animation_builder.ts]: No comma. Fix typo: A set of options that can include a time delay and additional developer-defined parameters.

File: packages/animations/src/animation_event.ts [hunk 4]
```diff
@@ -38,11 +38,33 @@
  * @experimental Animation support is experimental.
```
Comment (jenniferfell) [inline packages/animations/src/animation_event.ts]: Animation support in general isn't. Is this specific interface still experimental?

File: packages/animations/src/players/animation_group_player.ts [hunk 5]
```diff
@@ -9,6 +9,14 @@
 import {scheduleMicroTask} from '../util';
 import {AnimationPlayer} from './animation_player';
 
+/**
+ * A programmatic controller for a group of reusable animations.
+ * Used internally to control animations.
```
Comment (jenniferfell) [inline packages/animations/src/players/animation_group_player.ts]: Good note!

File: packages/animations/src/players/animation_player.ts [hunk 6]
```diff
@@ -8,37 +8,112 @@
...
+   * Ends the animation, invoking the `onDone()` callback.
+   */
   finish(): void;
+  /**
+   * Destroys the animation, after invoking the `beforeDestroy()` callback.
+   * Calls the `onDestroy()` callback when destruction is completed.
+   */
   destroy(): void;
+  /**
+   * Resets the animation to it initial state.
```
Comment (jenniferfell) [inline packages/animations/src/players/animation_player.ts]: it's

File: packages/animations/src/players/animation_player.ts [hunk 7]
```diff
@@ -8,37 +8,112 @@
...
+   */
   reset(): void;
+  /**
+   * Sets the position of the animation.
+   * @param p A point in screen coordinates?? (no one-letter param names allowed now)
```
Comment (jenniferfell) [inline packages/animations/src/players/animation_player.ts]: make sure to get an answer

File: packages/animations/src/players/animation_player.ts [hunk 8]
```diff
@@ -8,37 +8,112 @@
...
+   */
   setPosition(p: any /** TODO #9100 */): void;
+  /**
+   * Reports the current position of the animation.
+   * @returns The position. ??(what does the number mean?)
```
Comment (jenniferfell) [inline packages/animations/src/players/animation_player.ts]: I like ?? for notes to reviewers. Easy to search and remove before approval review.

File: packages/animations/src/players/animation_player.ts [hunk 9]
```diff
@@ -8,37 +8,112 @@
...
+   */
   getPosition(): number;
+  /**
+   * Retrieves the parent of this player.
+   * @return The parent player, if any. ?? (when/why does it have a parent?)
```
Comment (jenniferfell) [inline packages/animations/src/players/animation_player.ts]: This text doesn't seem to be captured in the output doc

File: packages/animations/src/players/animation_player.ts [hunk 10]
```diff
@@ -8,37 +8,112 @@
...
+   */
   parentPlayer: AnimationPlayer|null;
+  /**
+   * Reports the total run time of the animation.
+   * @returns The time in milliseconds. ??
```
Comment (jenniferfell) [inline packages/animations/src/players/animation_player.ts]: Again, the @returns for properties doesn't seem to appear in the doc that is generated.

**Review by jenniferfell — CHANGES_REQUESTED** [review]

File: packages/animations/src/players/animation_player.ts (hunk 10 above)
Comment (jbogarthyde) [reply packages/animations/src/players/animation_player.ts]: my fault -- these are properties, not methods. They have a value, not a return value.

File: packages/animations/src/players/animation_player.ts (hunk 8 above)
Comment (jbogarthyde) [reply packages/animations/src/players/animation_player.ts]: Thanks for the review -- I have made your changes.
Still need technical review and resolution of ?? questions from @matsko
 (and eng help with the experimental status/golden file).

File: packages/animations/src/players/animation_player.ts (hunk 7 above)
Comment (matsko) [reply packages/animations/src/players/animation_player.ts]: milliseconds

File: packages/animations/src/players/animation_player.ts (hunk 8 above)
Comment (matsko) [reply packages/animations/src/players/animation_player.ts]: The current amount of milliseconds as to where the animation is playing.

File: packages/animations/src/animation_builder.ts [hunk 11]
```diff
@@ -9,54 +9,71 @@ import {AnimationMetadata, AnimationOptions} from './animation_metadata';
...
  *
- * The purpose if this service is to produce an animation sequence programmatically within an
- * angular component or directive.
+ * @usageNotes
  *
- * Programmatic animations are first built and then a player is created when the build animation is
- * attached to an element.
+ * To use this service, add it to your component or directive as a dependency.
+ * The service is instantiated along with your component. Access the instance
+ * through the `_builder` property.
```
Comment (matsko) [inline packages/animations/src/animation_builder.ts]: `_builder` implies it's private. Please review this sentence.

**Review by matsko — CHANGES_REQUESTED** [review]

**Review by matsko — APPROVED** [review]

File: packages/animations/src/animation_event.ts (hunk 4 above)
Comment (matsko) [reply packages/animations/src/animation_event.ts]: Let's keep it as `@experimental` for now. Once we approach NG7 we can figure out if/when that will be removed.

Comment (jbogarthyde) [reply packages/animations/src/animation_event.ts]: Still have to check that golden file matches current stable APIs

**Comment by kara** [comment]
@jbogarthyde It looks like this PR Is still missing approval from @jenniferfell and the public-api group. When all the approvals are in, I can help get this merged.

**Review by IgorMinar — APPROVED** [review]

**Comment by IgorMinar** [comment]
This PR looks good to me. @jenniferfell's review is the last holding it back, but I think all her comments have been addressed already. If so, can you please dismiss/update the review and send this for merge? thanks

File: packages/animations/src/animation_builder.ts [hunk 12]
```diff
@@ -9,54 +9,72 @@ import {AnimationMetadata, AnimationOptions} from './animation_metadata';
...
  *
- * Programmatic animations are first built and then a player is created when the build animation is
- * attached to an element.
+ * To use this service, add it to your component or directive as a dependency.
+ * The service is instantiated along with your component.
+ *
+ * Apps do not typically need to create their own animation players, but if you
+ * do need to, follow these steps:
+ *
+ * * Use the `build()` method to create a programmatic animation using the
```
Comment (jenniferfell) [inline packages/animations/src/animation_builder.ts]: Please change to numbered steps. Use markdown syntax.

1. Use the `build()` method to...
1. Use the factory object to ...
1. Use the playter object to...

File: packages/animations/src/animation_builder.ts [hunk 13]
```diff
@@ -9,54 +9,72 @@ import {AnimationMetadata, AnimationOptions} from './animation_metadata';
...
- * AnimationBuilder.build}.
+ * A factory object returned from the `AnimationBuilder`.`build()` method.
  *
- * @experimental Animation support is experimental.
  */
 export abstract class AnimationFactory {
+  /**
+   * Creates an `AnimationPlayer` instance for the reusable animation defined by
+   * the `AnimationBuilder`.`build()` method that created this factory,
+   * and attaches the new player a DOM element.
```
Comment (jenniferfell) [inline packages/animations/src/animation_builder.ts]: Feels like a word is missing. "...and attaches the new player to a DOM element." ?

**Review by jenniferfell — CHANGES_REQUESTED** [review]

File: packages/animations/src/animation_builder.ts (hunk 13 above)
Comment (jbogarthyde) [reply packages/animations/src/animation_builder.ts]: Creates and attaches the player -- just a long clause about how it creates the player.

**Comment by jbogarthyde** [comment]
All green - please run G3 if needed.

**Comment by IgorMinar** [comment]
I wasn't sure if the new jsdoc tags could affect g3 in some way since closure reads them, so I kicked off a presubmit at http://test/OCL:208541253:BASE:208541646:1534195208651:b2b43c02

**Review by jenniferfell — APPROVED** [review]

**Comment by benlesh** [comment]
Forced google3 status. Will merge in a few.

**Pull request closed without merging**